src/AppInstallerCommonCore/Runtime.cpp

```diff
@@ -353,6 +353,33 @@ namespace AppInstaller::Runtime
         return wil::test_token_membership(nullptr, SECURITY_NT_AUTHORITY, SECURITY_BUILTIN_DOMAIN_RID, DOMAIN_ALIAS_RID_ADMINS);
     }
 
+    bool IsNTFS(const std::filesystem::path& filePath)
+    {
+        wil::unique_hfile fileHandle{ CreateFileW(
+            filePath.c_str(), /*lpFileName*/
+            GENERIC_READ, /*dwDesiredAccess*/

```

Comment:
>GENERIC_READ [](start = 12, length = 12)

I think we could ask for even less access than this to improve usability outside of this specific scenario.

Reply:
[https://docs.microsoft.com/en-us/windows/win32/secauthz/generic-access-rights](https://docs.microsoft.com/en-us/windows/win32/secauthz/generic-access-rights)

Constant Generic meaning
GENERIC_ALL All possible access rights
GENERIC_EXECUTE Execute access
GENERIC_READ Read access
GENERIC_WRITE Write access

I think GENERIC_READ is the least access

---
In reply to: [510499353](https://github.com/microsoft/winget-cli/pull/625#discussion_r510499353) [](ancestors = 510499353)

Reply:
Of the GENERIC ones, sure, but FILE_READ_ATTRIBUTES is certainly less than GENERIC_READ, as it is one part of it.

[https://docs.microsoft.com/en-us/windows/win32/fileio/file-security-and-access-rights](https://docs.microsoft.com/en-us/windows/win32/fileio/file-security-and-access-rights)

---
In reply to: [510523433](https://github.com/microsoft/winget-cli/pull/625#discussion_r510523433) [](ancestors = 510523433,510499353)